src/test/compile-fail/empty-struct-braces-expr.rs

```diff
@@ -29,6 +29,6 @@ fn main() {
 
     let xe1 = XEmpty1; //~ ERROR expected value, found struct `XEmpty1`
     let xe1 = XEmpty1(); //~ ERROR expected function, found struct `XEmpty1`
-    let xe3 = XE::Empty3; //~ ERROR no associated item named `Empty3` found for type
-    let xe3 = XE::Empty3(); //~ ERROR no associated item named `Empty3` found for type
+    let xe3 = XE::Empty3; //~ ERROR no variant named `Empty3` found for type
```

Comment:
In this context nothing suggests for sure that `Empty3` is a variant and not an associated constant or method.
At the same time, in other contexts like tuple struct patterns `match { E::X(..) => {} }` we can say for sure that `X` is a variant, but in this PR diagnostics are not context-dependent, only guessing based on case etc is used.
What I'd want to see is a context-dependent message worded and formatted uniformly with other "expected, (not) found" resolution diagnostics.
This is a bit vague and this PR is still an improvement, so I won't block this PR on these changes.

(I'm basically adding  a "rewrite this from scratch" item in my TODO list now.)